pkg/cmd/deadman/operator_condition_challenger.go

```diff
@@ -0,0 +1,179 @@
+	configv1lister "github.com/openshift/client-go/config/listers/config/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type OperatorConditionChallenger struct {
+	// operatorToTimeToCheckForStaleness is a map from operator name to the time at which the condition messages will be
+	// reset to a "checking for activity" message.
+	operatorToTimeToCheckForStaleness map[string]time.Time
+
+	durationAllowedBetweenStalenessChecks time.Duration
+
+	clusterOperatorClient configv1versionedtypes.ClusterOperatorInterface
+	clusterOperatorLister configv1lister.ClusterOperatorLister
+
+	queue workqueue.RateLimitingInterface
+}
+
+// NewOperatorConditionChallenger is a controller that sets operator condition messages to ensure that the operator
+// in question updates the condition messages back.  If the condition messages are not changed, then the operator
+// condition is considered stale.
+func NewOperatorConditionChallenger(
+	configClient configeversionedclient.Interface,
+	configInformers configv1informers.SharedInformerFactory,
+	durationAllowedBetweenStalenessChecks time.Duration,
+) *OperatorConditionChallenger {
+
+	c := &OperatorConditionChallenger{
+		operatorToTimeToCheckForStaleness:     map[string]time.Time{},
+		durationAllowedBetweenStalenessChecks: durationAllowedBetweenStalenessChecks,
+
+		clusterOperatorClient: configClient.ConfigV1().ClusterOperators(),
+		clusterOperatorLister: configInformers.Config().V1().ClusterOperators().Lister(),
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "OperatorConditionChallenger"),
+	}
+
+	configInformers.Config().V1().ClusterOperators().Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				clusterOperator := obj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				clusterOperator := newObj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+		},
+		1*time.Minute,
+	)
+
+	return c
+}
+
+func getMostRecentConditionTransitionTime(clusterOperator *configv1.ClusterOperator) *time.Time {
+	if len(clusterOperator.Status.Conditions) == 0 {
+		return nil
+	}
+
+	newestTime := time.Time{}
+	for _, condition := range clusterOperator.Status.Conditions {
+		if condition.LastTransitionTime.After(newestTime) {
+			newestTime = condition.LastTransitionTime.Time
+		}
+	}
+	return &newestTime
+}
+
+func (c *OperatorConditionChallenger) syncHandler(ctx context.Context, key string) error {
+	clusterOperator, err := c.clusterOperatorLister.Get(key)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if isCheckingForStaleness(clusterOperator) {
```

Comment:
Do we need to handle the case when the operator is unmanaged?

Reply:
> Do we need to handle the case when the operator is unmanaged?

Meaning you're concerned that an unmanaged operator wouldn't set status?  Seems like it would need to continually set an "I'm not sure" status, so it would be a bug either way.

Reply:
I'm not sure, but if you set Managed=False, we go upgradeable=False... In the past, people used managed=false to disable operators completely (for whatever reason). I'm asking if this should be reflected in the status/message (iow. "checking for stale status" vs. "operator unmanaged, status is stale, waiting for it to come back")

Reply:
the data shouldn't be stale because the operator should still be setting *something* indicating that it isn't doing work.